traits/observation/tests/test_observe.py

```diff
@@ -653,3 +655,74 @@ def test_apply_observers_different_target(self):
         # then
         # the handler should be called twice as the targets are different.
         self.assertEqual(handler.call_count, 2)
+
+
+# ---- Low-level tests for async dispatch_same ------------------------------
+
+
+class TestAsyncDispatchSame(unittest.IsolatedAsyncioTestCase):
+    """Test low-level async dispatch."""
+
+    def setUp(self):
+        from traits.observation.observe import _active_handler_tasks
+
+        # ensure no lingering references to handler tasks after test run
+        self.addCleanup(_active_handler_tasks.clear)
+
+        push_exception_handler(reraise_exceptions=True)
+        self.addCleanup(pop_exception_handler)
+
+    async def test_async_dispatch(self):
+        event = []
+
+        async def handler(event):
+            await asyncio.sleep(0)
+            event.append('called')
+        
+        dispatch_same(handler, event)
+
+        await asyncio.sleep(0.1)
```

Comment:
Can we rework this (and similar tests) to wait on an event instead? That is, create an event with `handled = asyncio.Event()`, then do a `handled.set()` within the handler, and replace this line with something like `await asyncio.wait(handled.wait(), SOME_TIMEOUT)`. Then `SOME_TIMEOUT` can be quite large (e.g., 60 seconds), since we don't expect to hit the timeout in a normal test run.